mount-daemon/main.go

```diff
@@ -176,90 +160,129 @@ func createManager(opts *options) (*managerConfig, error) {
 		}
 
-		tlsClientConfig := rest.TLSClientConfig{}
-		tlsClientConfig.CAFile = opts.certFile
+		return ctrl.Result{}, false, nil
+	}
 
-		config = &rest.Config{
-			Host:            "https://" + net.JoinHostPort(opts.host, opts.port),
-			TLSClientConfig: tlsClientConfig,
-			BearerToken:     string(token),
-			BearerTokenFile: opts.tokenFile,
+	if updated := nnfcontroller.InitializeClientMountStatus(clientMount); updated {
+		if err = statusUpdate(clnt, clientMount, clientLog); err != nil {
+			clientLog.Error(err, "unable to update status after initializing the clientmount status")
 		}
+		return ctrl.Result{Requeue: true}, false, nil
 	}
 
-	return &managerConfig{config: config, namespace: opts.name, mock: opts.mock, timeout: opts.timeout}, nil
-}
+	// Add finalizer if it doesn't exist
+	if !controllerutil.ContainsFinalizer(clientMount, finalizerClientMount) {
+		controllerutil.AddFinalizer(clientMount, finalizerClientMount)
+		if err := clnt.Update(context.TODO(), clientMount); err != nil {
+			if apierrors.IsConflict(err) {
+				clientLog.Error(err, "conflict while setting finalizer")
+			} else {
+				clientLog.Error(err, "unable to set finalizer")
+			}
+			clientMount.Status.SetResourceErrorAndLog(err, clientLog)
+			if suErr := statusUpdate(clnt, clientMount, clientLog); suErr != nil {
+				clientLog.Error(suErr, "unable to update finalizer error status")
+			}
+			return ctrl.Result{Requeue: true}, false, nil
+		}
+		clientLog.Info("updated finalizer")
+		return ctrl.Result{Requeue: true}, false, nil
+	}
 
-func startManager(config *managerConfig) {
-	setupLog.Info("GOMAXPROCS", "value", runtime.GOMAXPROCS(0))
+	mountChanged := false
+	nnfcontroller.ChangeMountAllPre(clientMount)
+	if mountChanged, err = nnfcontroller.ChangeMountAll(context.TODO(), clnt, clientMount, clientMount.Spec.DesiredState, setupLog); err != nil {
+		resourceError := dwsv1alpha2.NewResourceError("mount/unmount failed").WithError(err)
+		clientMount.Status.SetResourceErrorAndLog(resourceError, setupLog)
+	} else {
+		nnfcontroller.ChangeMountAllPost(clientMount)
+	}
+	if err = statusUpdate(clnt, clientMount, clientLog); err != nil {
+		clientLog.Error(err, "unable to update status after ChangeMountAll")
+		return ctrl.Result{Requeue: true}, mountChanged, nil
+	}
 
-	namespaceCache := make(map[string]cache.Config)
-	namespaceCache[config.namespace] = cache.Config{}
-	namespaceCache[corev1.NamespaceDefault] = cache.Config{}
+	return ctrl.Result{}, mountChanged, nil
+}
 
-	mgr, err := ctrl.NewManager(config.config, ctrl.Options{
-		Scheme:         scheme,
-		LeaderElection: false,
-		Cache:          cache.Options{DefaultNamespaces: namespaceCache},
-	})
-	if err != nil {
-		setupLog.Error(err, "unable to start manager")
-		os.Exit(1)
+func reconcileAll(clnt client.Client, namespace string) (ctrl.Result, error) {
+	clientMountList := &dwsv1alpha2.ClientMountList{}
+	listOptions := []client.ListOption{
+		client.InNamespace(namespace),
```

Comment:
I know flux shouldn't be running multiple workflows on the same compute, but should this be limited to a workflow along with the namespace?

What would happen if there's another clientmount present from an old workflow that didn't clean up and that returns an error? Would this logic find that error and bomb out without doing any work on the _actual_ clientmount that was requested?